### PR TITLE
samples: ipc_service: Adapt sample to cache and sysbuild

### PR DESCRIPTION
**Pull request by KAGA164**
This adapts and align samples to sysbuild and cache support. The OpenAMP backend is no longer support
on nRF54h20 target due to cache issues and it is
not intended to being used on this platform.

### CODE REVIEW

**Review by grochu — APPROVED** [review]
The changes look good to me.

File: samples/ipc/ipc_service/README.rst [hunk 1]
```diff
@@ -13,7 +13,14 @@ Overview
...
 The sample application tests throughput of the IPC service with available backends.
-Currently, the sample supports OpenAMP RPMSG and ICMSG backends.
+Currently, the sample supports the following backends:
+
+* the :ref:`zephyr:nrf5340dk_nrf5340` board
+  * the `OpenAMP`_ library
+  * :ref:`zephyr:ipc_service_backend_icmsg`
+
+* the :ref:`zephyr:nrf54h20dk_nrf54h20` board:
+  * :ref:`zephyr:ipc_service_backend_icbmsg`
```
Comment (maje-emb) [inline samples/ipc/ipc_service/README.rst:23]:
```suggestion
Currently, the sample supports the following backends:

* the :ref:`zephyr:nrf5340dk_nrf5340` board

  * the `OpenAMP`_ library
  
  * :ref:`zephyr:ipc_service_backend_icmsg`

* the :ref:`zephyr:nrf54h20dk_nrf54h20` board:

  * :ref:`zephyr:ipc_service_backend_icbmsg`
```
It seems to me that there should be blank line between the bullets.

File: samples/ipc/ipc_service/README.rst [hunk 2]
```diff
@@ -82,17 +89,16 @@ Use these overlays when building the IPC sample to test the following scenarios:
 
 **nRF54H20 DK**
 
-You can build the sample to test IPC between the application and PPR core using the ICMSG backend, as follows:
+You can build the sample to test IPC between the application and PPR core using the ICBMSG backend, as follows:
```
Comment (maje-emb) [inline samples/ipc/ipc_service/README.rst:92]:
```suggestion
You can build the sample to test IPC between the application and PPR core using the ``icbmsg`` backend, as follows:
```
In other places we use lower case latters.
Or consider adding a reference to `ipc_service_backend_icbmsg`.
@peknis what do you think about it?

File: samples/ipc/ipc_service/README.rst [hunk 3]
```diff
@@ -82,17 +89,16 @@ Use these overlays when building the IPC sample to test the following scenarios:
...
 
 .. code-block:: console
 
    west build -p -b nrf54h20dk_nrf54h20_cpuapp -T sample.ipc.ipc_service.nrf54h20dk_cpuapp_cpuppr_icmsg .
 
-You can build the sample to test IPC between the application and radio domains using either the RPMsg or the ICMSG backend, as follows:
+You can build the sample to test IPC between the application and radio domains using the ICBMSG backend, as follows:
```
Comment (maje-emb) [inline samples/ipc/ipc_service/README.rst:98]: As above.

File: samples/ipc/ipc_service/README.rst (hunk 2 above)
Comment (peknis) [reply samples/ipc/ipc_service/README.rst:92]: I'd prefer having the link.

File: samples/ipc/ipc_service/README.rst (hunk 1 above)
Comment (peknis) [reply samples/ipc/ipc_service/README.rst:23]: Same level bullets do not need a blank line in between.

**Review by peknis — APPROVED** [review]

File: samples/ipc/ipc_service/README.rst (hunk 2 above)
Comment (KAGA164) [reply samples/ipc/ipc_service/README.rst]: I agree. Changed to link

File: samples/ipc/ipc_service/README.rst (hunk 1 above)
Comment (KAGA164) [reply samples/ipc/ipc_service/README.rst]: Done

**Review by maje-emb — APPROVED** [review]

File: samples/ipc/ipc_service/sample.yaml [hunk 4]
```diff
@@ -15,34 +15,34 @@ tests:
...
 
   sample.ipc.ipc_service.nrf5340dk_rpmsg_cpuapp_sending:
     platform_allow:
       - nrf5340dk_nrf5340_cpuapp
     integration_platforms:
       - nrf5340dk_nrf5340_cpuapp
     extra_configs:
-      - ipc_service_CONFIG_APP_IPC_SERVICE_SEND_INTERVAL=1
+      - CONFIG_APP_IPC_SERVICE_SEND_INTERVAL=1
       - remote_CONFIG_APP_IPC_SERVICE_SEND_INTERVAL=200000000
```
Comment (gchwier) [inline samples/ipc/ipc_service/sample.yaml:28]: move all `remote_CONFIG*` from `extra_configs` to `extra_args` to pass Twister

Comment (KAGA164) [reply samples/ipc/ipc_service/sample.yaml]: Ok. Done. Will it be fixed later in twister?

Comment (gchwier) [reply samples/ipc/ipc_service/sample.yaml]: I don't know. In docs is written `Extra configuration options to be merged with a master prj.conf `, maybe should be extended to support also sysbuild or some extra clarification could be added. For now using `extra_args` is enough

**Comment by gchwier** [comment]
probable you need also to align with Hardware Model V2, it will be merged soon: https://github.com/nrfconnect/sdk-nrf/pull/14626

**Comment by cvinayak** [comment]
@shanthanordic i missed your request for review.... hope this has not caused damage?

**Pull request merged**